### DISCUSSION TITLE
✨ Allow setting resource limit to pipelines

### DISCUSSION THREAD

**Discussion by marcosnils** [dagger/dagger#12520] [Ideas]
### What are you trying to do?

This feature has been requested by the user in Discord **YourTechBud**

" Using dagger to build some code and images. I want to restrict the cpu and memory allocated to this. How can I do that?"

cc @YourTechBud
 

### Why is this important to you?

N/A

### How are you currently working around this?

N/A

**Comment by sipsma** [comment]
just a note: buildkit has support for specifying an existing cgroup for an exec to run in. That by itself is not the greatest dx because it requires the cgroup to be created beforehand manually, but I feel like we could do something with our shim to improve this further, maybe make cgroups on the fly so that users can just specify actual cpu/mem limitations?

**Comment by marcosnils** [comment]
> but I feel like we could do something with our shim to improve this further, maybe make cgroups on the fly so that users can just specify actual cpu/mem limitations?

that'd be awesome.

idea: we could inspire on the approach docker run` uses for setting CPU and memory constraints? After all, users are already familiarized with those.
